pkg/plugins/document/dynamodb/dynamodb.go

```diff
@@ -121,28 +157,46 @@ func (s *DynamoDocService) Set(key *document.Key, value map[string]interface{})
 
 	_, err = s.client.PutItem(input)
 	if err != nil {
-		return err
+		return newErr(
+			codes.Internal,
+			"error putting item",
+			err,
+		)
 	}
 
 	return nil
 }
 
 func (s *DynamoDocService) Delete(key *document.Key) error {
+	newErr := errors.ErrorsWithScope("DynamoDocService.Delete")
 	err := document.ValidateKey(key)
+
 	if err != nil {
-		return err
+		return newErr(
+			codes.InvalidArgument,
+			"invalid key",
+			err,
+		)
 	}
 
 	keyMap := createKeyMap(key)
 	attributeMap, err := dynamodbattribute.MarshalMap(keyMap)
 	if err != nil {
-		return fmt.Errorf("failed to marshal keys: %v", key)
+		return newErr(
+			codes.InvalidArgument,
+			fmt.Sprintf("failed to marshal keys: %v", key),
+			err,
+		)
 	}
 
 	tableName, err := s.getTableName(*key.Collection)
 
 	if err != nil {
-		return err
+		return newErr(
+			codes.NotFound,
+			"unable to find table",
```

Comment:
was thinking about this, do you think it would be appropriate to include non-sensitive args in the error factory for scope?

update errors.ErrorsWithScope to be
```go
func ErrorsWithScope(scope string, args ...interface{})
```

e.g. for not finding topic: `newErr := errors.ErrorsWithScope("SnsEventService.Publish", topic)`

and print them by default in error messages? so provided args will be included with error message by default?

Reply:
Yes, I think so. Its really useful from an operational support perspective.  I thinking about this now the adaptor layer is probably the best spot to do this.